cmake/SearchForStuff.cmake

```diff
@@ -679,7 +689,11 @@ endif ()
 
 ########################################
 # Find gdal
-include (FindGDAL)
+find_package(GDAL CONFIG)
+if ("${GDAL_LIBRARY}" STREQUAL "")
+  unset(GDAL_FOUND CACHE)
+  include (FindGDAL)
+endif()
```

Comment:
Is this necessary for the PR to work? Can't we just do `find_package(GDAL CONFIG)` if `include(FindGDAL)` does not find GDAL?

Reply:
> Is this necessary for the PR to work? Can't we just do `find_package(GDAL CONFIG)` if `include(FindGDAL)` does not find GDAL?

I think it will not work for this PR. 
Do you want to try it and see?

Reply:
Yes, please. As I already clarified earlier in the review, we try to be conservative, and only add new ways of finding libraries only in the case old ways do not work. If there is a problem in that case, we can debug.

Reply:
ok. I am trying.

Reply:
It seems that this version is working.  Given this, are you sure that the:
~~~
if ("${GDAL_LIBRARY}" STREQUAL "")
  unset(GDAL_FOUND CACHE)
  find_package(GDAL CONFIG)
endif()
~~~
is actually necessary?

Reply:
I will remove it.
Any other comments?
I will work on it on the evening.

Reply:
Thanks, for the future if we can test if changes are actually necessary before submitting the PR it can be helpful to avoid reducing the time spent in review, thanks!

Reply:
> Thanks, for the future if we can test if changes are actually necessary before submitting the PR it can be helpful to avoid reducing the time spent in review, thanks!

I already have that PR with minimal changes. But the repo maintainer didn't respond, So I continue to developing, Also expend to x86 that now is working, also combined it with some Linux fixes. 

So for the future if we have the same response rate like now, I think it will be ok.

If you have more comments for now that you think that something is not necessary, please let me know on that spot that I will fix everything this evening. Thank you! :)

Reply:
> I already have that PR with minimal changes. But the repo maintainer didn't respond, So I continue to developing,

If you want to continue developmenent to add more feature on top of a PR, you can just create a new branch out of the PR and add more work there.

Reply:
This is already the new branch with the new PR.
The old branch and old PR not relevant anymore, and I close it for your request.

Reply:
done.